app/pulp/app/models/task.py

```diff
@@ -205,6 +206,22 @@ def set_failed(self, exc, einfo):
         self.result = exception_to_dict(exc, einfo.traceback)
         self.save()
 
+    def cancel(self):
+        """
+        Set the task to a canceled state if it is in an incomplete state.
+
+        This updates the :attr: `finished_at` attribute, sets the :attr: `state` to
+        :attr: `CANCELED`
+
+        """
+        if self.state not in TASK_INCOMPLETE_STATES:
+            self.state = Task.CANCELED
+            self.finished_at = timezone.now()
+            self.save()
```

Comment:
This is approximately where a call should be made to celery to actually cancel the task. It would be a good idea to track that somehow either via a TODO here or by making a task in redmine. It might also be an easy 1-line addition to just knock out, but I'll leave that up to you.